h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -2339,6 +2357,49 @@ else if (gaussian.equals(_parms._family) && Link.identity.equals(_parms._link))
       }
     }
 
+    /***
+     * Estimate dispersion factor using maximum likelihood.  I followed section IV of the doc in 
+     * https://h2oai.atlassian.net/browse/PUBDEV-8683 . 
+     */
+    public double estimateMLSE(GLMTask.ComputeMLSETsk mlCT, double alpha, double[] beta) {
+      double constantValue = mlCT._wsum + mlCT._sumlnyiOui - mlCT._sumyiOverui;
+      DataInfo dinfo = _state.activeData();
+      Frame adaptedF = dinfo._adaptedFrame;
+      int count = 0;
+      long currTime = System.currentTimeMillis();
+      long modelBuiltTime = currTime - _model._output._start_time;
+      
+      // stopping condition for while loop are:
+      // 1. magnitude of iterative change to se < EPS
+      // 2. for every 100th iteration, we check for additional stopping condition:
+      //    a.  User requests stop via stop_requested;
+      //    b.  time taken to find the dispersino parameter has exceeded time taken to build the GLM model.
+      while (true) {
+        GLMTask.ComputeDiTriGammaTsk ditrigammatsk = new GLMTask.ComputeDiTriGammaTsk(null, dinfo, _job._key, beta,
```

Comment:
this could probably be optimized, it doesn't seem like you need the whole row - only response, offset and weight is needed, correct?

repeated extraction of every row of the dataset will be expensive